crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -152,3 +234,139 @@ impl PersistedQueriesOperationCounts {
+                id: id_two.to_string(),
+                body: body_two.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn relay_manifest_with_anonymous_operations_fails() {
+        let relay_manifest = r#"{
+      "ed145403db84d192c3f2f44eaa9bc6f9": "query {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n"
+    }"#;
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest_result: Result<ApolloPersistedQueryManifest, RoverClientError> =
+            relay_manifest.try_into();
+        assert!(matches!(
+            apollo_manifest_result,
+            Err(RoverClientError::RelayOperationParseFailures { .. })
+        ));
+    }
+
+    #[test]
+    fn relay_manifest_with_anonymous_operation_and_valid_operation_fails() {
+        let relay_manifest = r#"{
+      "ed145403db84d192c3f2f44eaa9bc6f9": "query {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n",
+      "alskdjlasj": "query NamedQuery { topStory }"
+    }"#;
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest_result: Result<ApolloPersistedQueryManifest, RoverClientError> =
+            relay_manifest.try_into();
+        assert!(matches!(
+            apollo_manifest_result,
+            Err(RoverClientError::RelayOperationParseFailures { .. })
+        ));
+    }
+
+    #[test]
+    fn relay_manifest_with_invalid_operation_type_succeeds() {
+        let id = "ed145403db84d192c3f2f44eaa9bc6f9";
+        let body = "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n";
+        let relay_manifest = serde_json::json!({id: body}).to_string();
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest: ApolloPersistedQueryManifest = relay_manifest
+            .try_into()
+            .expect("could not convert relay manifest to apollo manifest");
+        assert_eq!(
+            apollo_manifest.operations[0],
+            PersistedQueryOperation {
+                name: "NewsfeedQuery".to_string(),
+                r#type: PersistedQueryOperationType::Query,
+                id: id.to_string(),
+                body: body.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn empty_relay_manifest_cannot_be_converted() {
```

Comment:
Not convinced that this should be an error, but seems reasonable enough. If there's some valid use case for publishing empty relay manifests we can relax the error later.